src/foam/dao/FixedSizeDAO.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'FixedSizeDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: function () {/*
+    DAO Decorator that stores a fixed number of objects.
+  */},
+
+  javaImports: [
+    'org.apache.commons.collections.buffer.CircularFifoBuffer',
+    'foam.dao.Sink',
+  ],
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'fixedArraySize',
+      value: 10000
+    },
+    {
+      class: 'Object',
+      name: 'FixedSizeArray',
+      javaType: 'org.apache.commons.collections.buffer.CircularFifoBuffer',
+      javaFactory: `return new org.apache.commons.collections.buffer.CircularFifoBuffer(getFixedArraySize()); `
```

Comment:
This buffer is itself not thread safe. 
Two options: 
1. Buffer fifo = BufferUtils.synchronizedBuffer(new CircularFifoBuffer());
2. Use a regular Array and add a synchronized 'nextIndex' method. 

Option 1 is easier as the CircularFifoBuffer handles correct iteration order during toArray on select().